client/mapreduce/src/main/java/org/apache/accumulo/mapreduce/impl/InputInfoImpl.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.sample.SamplerConfiguration;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.util.Pair;
+import org.apache.accumulo.mapreduce.InputInfo;
+import org.apache.hadoop.io.Text;
+
+public class InputInfoImpl implements InputInfo {
+  String tableName;
+  ClientInfo clientInfo;
+  Authorizations scanAuths;
+  String context;
+  String logLevel;
+  Collection<Range> ranges;
+  Collection<Pair<Text, Text>> cfcqPairs;
+  IteratorSetting cfg;
+  SamplerConfiguration samplerConfig;
+  Map<String, String> hints;
+
+  public InputInfoImpl(String tableName, ClientInfo clientInfo, Authorizations scanAuths, String context,
+                       String logLevel, Collection<Range> ranges, Collection<Pair<Text, Text>> cfcqPairs, IteratorSetting cfg,
+                       SamplerConfiguration samplerConfig, Map<String, String> hints) {
+    this.tableName = tableName;
+    this.clientInfo = clientInfo;
+    this.scanAuths = scanAuths;
+    this.context = context;
+    this.logLevel = logLevel;
+    this.ranges = ranges;
+    this.cfcqPairs = cfcqPairs;
+    this.cfg = cfg;
+    this.samplerConfig = samplerConfig;
+    this.hints = hints;
+  }
+
+  @Override
+  public String getTableName() {
+    return tableName;
+  }
+
+  @Override
+  public ClientInfo getClientInfo() {
+    return clientInfo;
+  }
+
+  public Authorizations getScanAuths() {
+    return scanAuths;
+  }
+
+  @Override
+  public String getContext() {
+    return context;
+  }
+
+  @Override
+  public String getLogLevel() {
+    return logLevel;
+  }
+
+  @Override
+  public Collection<Range> getRanges() {
+    return ranges;
+  }
+
+  @Override
+  public Collection<Pair<Text, Text>> getCfcqPairs() {
+    return cfcqPairs;
+  }
+
+  @Override
+  public IteratorSetting getIterators() {
+    return cfg;
+  }
+
+  @Override
+  public SamplerConfiguration getSamplerConfig() {
+    return samplerConfig;
+  }
+
+  @Override
+  public Map<String, String> getHints() {
+    return hints;
+  }
+
+  public static class InputInfoBuilderImpl implements InputFormatBuilder, InputInfo.InputFormatBuilder.ClientParams, InputInfo.InputFormatBuilder.TableParams,
+  InputInfo.InputFormatBuilder.AuthsParams, InputInfo.InputFormatBuilder.InputFormatOptions {
+
+    String tableName;
+    ClientInfo clientInfo;
+    Authorizations scanAuths;
+
+    String context;
+    String logLevel;
+    Collection<Range> ranges;
+    Collection<Pair<Text, Text>> cfcqPairs;
+    IteratorSetting cfg;
+    SamplerConfiguration samplerConfig;
+    Map<String, String> hints;
+
+    @Override
+    public InputInfo.InputFormatBuilder.TableParams clientInfo(ClientInfo clientInfo) {
+      this.clientInfo = clientInfo;
+      return this;
+    }
+    @Override
+    public InputInfo.InputFormatBuilder.TableParams clientProperties(Properties clientPropsFiles) {
```

Comment:
The naming of `clientPropsFiles` should be changed to `clientProps` as there is no file.

However, you could add a `clientPropertiesFile` method and use the code in my PR #715 to get ClientInfo from the file path.